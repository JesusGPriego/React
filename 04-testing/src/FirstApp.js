//import React, { Fragment } from "react";
import React from "react";
import PropTypes from "prop-types";

const FirstApp = ({ greeting, subtitle }) => {
  const saludo = "Hola mundo";
  const persona = {
    nombre: "Jesús",
    edad: 28,
  };
  console.log(greeting);
  return (
    // <div>
    //   <h1>Hola mundo</h1>
    //   <p>Ésta es mi primera app</p>
    // </div>
    // <Fragment>
    //   <h1>Hola mundo</h1>
    //   <p>Ésta es mi primera app</p>
    // </Fragment>

    <>
      {/* <pre>{JSON.stringify(persona, null, 3)}</pre> */}
      {/* <h1>{saludo}</h1>*/}
      <h1>{greeting}</h1>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  greeting: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  greeting: "Hello world",
  subtitle: "I am a subtitle",
};

export default FirstApp;
