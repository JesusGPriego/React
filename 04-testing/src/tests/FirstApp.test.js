import { shallow } from "enzyme";
import FirstApp from "../FirstApp";
import React from "react";
import "@testing-library/jest-dom";
describe("Testing <FirstApp />", () => {
  //   test("should return 'Mahara'", () => {
  //     const greeting = "Mahara";
  //     const { getByText } = render(<FirstApp greeting="Mahara" />);
  //     expect(getByText(greeting)).toBeInTheDocument();
  //   });

  test("should show FirstApp correctly", () => {
    const greeting = "Mahara";
    const wrapper = shallow(<FirstApp greeting={greeting} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should show props subtitle", () => {
    const greeting = "Mahara";
    const subtitle = "I'm a subtitle";
    const wrapper = shallow(
      <FirstApp greeting={greeting} subtitle={subtitle} />
    );
    const subtitleText = wrapper.find("p").text();
    expect(subtitleText).toBe(subtitle);
  });
});
