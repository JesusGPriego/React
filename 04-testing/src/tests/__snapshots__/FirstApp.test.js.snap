// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FirstApp /> should show FirstApp correctly 1`] = `
<Fragment>
  <h1>
    Mahara
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
